src/Database/PostgreSQL/Consumers/Components.hs

```diff
@@ -325,14 +344,58 @@ spawnDispatcher ConsumerConfig{..} cs cid semaphore
           ]
         return (ccJobIndex job, Failed action)
 
+    updateJob :: (idx, Result) -> m ()
```

Comment:
Why a tuple instead of two parameters? Basically a nit, but anyway.

Reply:
Because it's modelled in the same way as `updateJobs` and binding the function is less clunky `... >>= updateJob` instead of `>>= uncurry updateJob`